core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -851,8 +852,16 @@ private[spark] class MapOutputTrackerWorker(conf: SparkConf) extends MapOutputTr
         var fetchedStatuses = mapStatuses.get(shuffleId).orNull
         if (fetchedStatuses == null) {
           logInfo("Doing the fetch; tracker endpoint = " + trackerEndpoint)
-          val fetchedBytes = askTracker[Array[Byte]](GetMapOutputStatuses(shuffleId))
-          fetchedStatuses = MapOutputTracker.deserializeMapStatuses(fetchedBytes, conf)
+          try {
+            val fetchedBytes = askTracker[Array[Byte]](GetMapOutputStatuses(shuffleId))
+            fetchedStatuses = MapOutputTracker.deserializeMapStatuses(fetchedBytes, conf)
+          } catch {
+            case e: IOException if
+            Throwables.getCausalChain(e).asScala.exists(_.isInstanceOf[BlockNotFoundException]) =>
```

Comment:
nit: indent.

Reply:
Shall we `logError` here?

Reply:
Maybe print logs at the `Executor` exception handling? I checked the code, seems it will not print any logs for `FetchFailedException` now?
https://github.com/apache/spark/blob/a22994333ae129a3edaac75101fb4217fe8881c7/core/src/main/scala/org/apache/spark/executor/Executor.scala#L618

Reply:
> Maybe print logs at the `Executor` exception handling?

I don't understand what do you mean by this.

> I checked the code, seems it will not print any logs for FetchFailedException now?

Yeah, but I think this's one is different. For me, I'd like to have a way to distinguish these two fetch failure.

Reply:
The `MetadataFetchFailedException` already contains the root cause message, I think we can just print logs when handling fetch failure exception at the `Executor` class?

Reply:
> The `MetadataFetchFailedException` already contains the root cause message

Oh, I miss that..then, it should be fine.